agent/src/runtime_connectors/runtime_facade.rs

```diff
@@ -107,39 +105,53 @@ impl<
             "Creating '{}' workload '{}' on agent '{}'",
             runtime.name(),
             workload_name,
-            agent_name
+            workload_spec.agent
         );
-
+        let (workload_channel, command_receiver) = WorkloadCommandChannel::new();
+        let workload_channel_retry = workload_channel.clone();
         tokio::spawn(async move {
-            let workload_name = workload_spec.name.clone();
-            let (workload_id, state_checker) = runtime
+            let instance_name = workload_spec.instance_name();
+            let create_result = runtime
                 .create_workload(
-                    workload_spec,
-                    control_interface_path,
+                    workload_spec.clone(),
+                    control_interface_path.clone(),
                     update_state_tx.clone(),
                 )
-                .await
-                .map_or_else(
-                    |err| {
-                        log::warn!("Failed to create workload: '{}': '{}'", workload_name, err);
-                        (None, None)
-                    },
-                    |(workload_id, state_checker)| (Some(workload_id), Some(state_checker)),
+                .await;
```

Comment:
I wonder if we can simplify things if this is also done in the control loop instead of here and the create command is queued in too.

Reply:
As agreed, let's leave it as is for the sake of readability.